### ISSUE TITLE
emcc: cannot find binaryen js libraries

### ISSUE THREAD

**Issue by Y0QIN** [emscripten-core/emscripten#8570]
emcc: cannot find binaryen js libraries (tried: ('/root/emsdk/binaryen/tag-1.38.22_32bit_binaryen/bin', '/root/emsdk/binaryen/tag-1.38.22_32bit_binaryen/share/binaryen'))

~/emsdk/binaryen/tag-1.38.22_32bit_binaryen/bin# ls -l
-rwxr-xr-x 1 root root 6647077  5月  9 15:42 asm2wasm
-rwxr-xr-x 1 root root 6424500  5月  9 15:42 wasm2js
-rwxr-xr-x 1 root root 6123266  5月  9 15:42 wasm-as
-rwxr-xr-x 1 root root 6324701  5月  9 15:41 wasm-ctor-eval
-rwxr-xr-x 1 root root 6117803  5月  9 15:42 wasm-dis
-rwxr-xr-x 1 root root 6128507  5月  9 15:42 wasm-emscripten-finalize
-rwxr-xr-x 1 root root 6213634  5月  9 15:42 wasm-merge
-rwxr-xr-x 1 root root 6216189  5月  9 15:42 wasm-metadce
-rwxr-xr-x 1 root root 6638577  5月  9 15:42 wasm-opt
-rwxr-xr-x 1 root root 6230195  5月  9 15:41 wasm-reduce
-rwxr-xr-x 1 root root 6352470  5月  9 15:42 wasm-shell

**Comment by Y0QIN** [comment]
The default is to find the 64-bit js：
emcc: cannot find binaryen js libraries (tried: ('/root/emsdk/binaryen/tag-1.38.22_64bit_binaryen/bin', '/root/emsdk/binaryen/tag-1.38.22_64bit_binaryen/share/binaryen'))
and I modified BINARYEN_ROOT to 32bit then occured this error also

**Comment by sbc100** [comment]
Hmm..  I don't see 1.38.32 in emsdk:

```
./emsdk update-tags
Fetching all tags from Emscripten Github repository...
Done. 159 tagged releases available, latest is 1.38.31.
Fetching all tags from Binaryen Github repository...
Done. 86 tagged Binaryen releases available, latest is 1.38.31.
```

How did you install this version?

**Comment by sbc100** [comment]
How did you install the sdk?

If I run `./emsdk install sdk-1.38.22-64bit` I see binaryen installed in `clang/e1.38.22_64bit/binaryen/bin`

**Comment by Y0QIN** [comment]
I run ./emsdk install latest and I see binaryen installed in clang/e1.38.22_32bit/binaryen/bin and  Build files have been written to: /root/emsdk/binaryen/tag-1.38.22_32bit_binaryen

**Comment by Y0QIN** [comment]
@sbc100
 I run ./emsdk install latest and I see binaryen installed in clang/e1.38.22_32bit/binaryen/bin and Build files have been written to: /root/emsdk/binaryen/tag-1.38.22_32bit_binaryen and I can‘t find binaryen js libraries in the path: /root/emsdk/binaryen/tag-1.38.22_64bit_binaryen/share/binaryen

**Comment by sbc100** [comment]
What does `./emsdk activate sdk-1.38.22-64bit` output?

**Comment by Y0QIN** [comment]
@sbc100 
since I don't install 64bit ,I only install 32bit
output is  following:
Writing .emscripten configuration file to user home directory /root/
Warning: The SDK/tool 'clang-e1.38.22-64bit' cannot be activated since it is not installed! Skipping this tool...
Warning: The SDK/tool 'node-8.9.1-64bit' cannot be activated since it is not installed! Skipping this tool...
Warning: The SDK/tool 'emscripten-1.38.22' cannot be activated since it is not installed! Skipping this tool...
Warning: The SDK/tool 'sdk-1.38.22-64bit' cannot be activated since it is not installed! Skipping this tool...

**Comment by Y0QIN** [comment]
@sbc100 maybe I should to install 64bit to solve this problem?

**Comment by sbc100** [comment]
What `./emsdk install` command did you use?  And can you show the output the `./emsdk activate` command that you used?

**Comment by Y0QIN** [comment]
@sbc100  I used ./emsdk install latest 
and use  ./emsdk activate latest
following output:
Writing .emscripten configuration file to user home directory /root/
The Emscripten configuration file /root/.emscripten has been rewritten with the following contents:

import os
LLVM_ROOT = '/root/emsdk/clang/tag-e1.38.22/build_tag-e1.38.22_32/bin'
NODE_JS = '/root/emsdk/node/8.9.1_32bit/bin/node'
EMSCRIPTEN_ROOT = '/root/emsdk/emscripten/tag-1.38.22'
EMSCRIPTEN_NATIVE_OPTIMIZER = '/root/emsdk/emscripten/tag-1.38.22_32bit_optimizer/optimizer'
BINARYEN_ROOT = '/root/emsdk/binaryen/tag-1.38.22_64bit_binaryen'
SPIDERMONKEY_ENGINE = ''
V8_ENGINE = ''
TEMP_DIR = '/tmp'
COMPILER_ENGINE = NODE_JS
JS_ENGINES = [NODE_JS]

To conveniently access the selected set of tools from the command line, consider adding the following directories to PATH, or call 'source ./emsdk_env.sh' to do this for you.

   /root/emsdk:/root/emsdk/clang/tag-e1.38.22/build_tag-e1.38.22_32/bin:/root/emsdk/node/8.9.1_32bit/bin:/root/emsdk/emscripten/tag-1.38.22:/root/emsdk/binaryen/tag-1.38.22_64bit_binaryen/bin
Set the following tools as active:
   clang-tag-e1.38.22-32bit
   node-8.9.1-32bit
   emscripten-tag-1.38.22-32bit
   binaryen-tag-1.38.22-32bit

**Comment by sbc100** [comment]
You're latest is very out-of-date.  The latest SDK should be 1.38.31.    I think if you run `emsdk update` and use a more recent version this problem is already fixes.

**Comment by Y0QIN** [comment]
@sbc100 
I remember that I have executed emsdk update before executing the install command. The result is still the same
Ok, let me try again.

**Comment by Y0QIN** [comment]
@sbc100 
I update again and Install 1.38.31.64bit.
Error downloading URL 'https://github.com/kripken/emscripten-fastcomp-clang/archive/1.38.31.tar.gz': IncompleteRead(7473 bytes read, 719 more expected)
Installation failed!
then I downloaded the packet myself and move to path：/root/emsdk/zips/
Then the installation is successful

But ,I meet a ImportError while executing emcc -v
Traceback (most recent call last):
  File "/root/emsdk/emscripten/1.38.31/emcc", line 6, in <module>
    from tools import python_selector
ImportError: No module named 'tools'

**Comment by sbc100** [comment]
Strange.  Looks like some kind of circular import issue with python.   What python version do you have installed?   I seem to remember there were issues with certain early versions of python3 in respect to this.

**Comment by Y0QIN** [comment]
@sbc100 my python version is python 3.4.3。
Maybe I should update my python to a recent version?

**Comment by sbc100** [comment]
yeah... looks like we depend on 3.5.   See https://www.quora.com/Are-circular-dependencies-allowed-in-python-3-5-but-not-in-3-4.   I'll add a check give a better warning message.

**Comment by Y0QIN** [comment]
I update my python to 3.7.3 and Install emsdk success 。
But there occurred exception :
Traceback (most recent call last):
  File "/root/emsdk/emscripten/1.38.31/emcc.py", line 3308, in <module>
    sys.exit(run(sys.argv))
  File "/root/emsdk/emscripten/1.38.31/emcc.py", line 477, in run
    code = run_process([shared.CLANG_CC, '-v'], check=False).returncode
  File "/root/emsdk/emscripten/1.38.31/tools/shared.py", line 163, in run_process
    ret = subprocess.run(cmd, check=check, input=input, *args, **kw)
  File "/opt/python3.7/lib/python3.7/subprocess.py", line 472, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/opt/python3.7/lib/python3.7/subprocess.py", line 775, in __init__
    restore_signals, start_new_session)
  File "/opt/python3.7/lib/python3.7/subprocess.py", line 1522, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
OSError: [Errno 8] Exec format error : '/root/emsdk/clang/e1.38.31_64bit/clang'

Why? Maybe I should upgrade my cmake?
